.github/workflows/compare_latest_version.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+import os
+import re
+from pathlib import Path
+from urllib.parse import urljoin
+
+import requests
+from bs4 import BeautifulSoup
+
+SNAPHU_URL = "https://web.stanford.edu/group/radar/softwareandlinks/sw/snaphu/"
+
+
+def fetch_stanford_version_url() -> str:
+    """
+    Find the link to the latest version of "snaphu-vXXX.tar.gz" from the given URL.
+
+    Parameters
+    ----------
+    url : str
+        The URL of the webpage to search for the snaphu tarball.
+
+    Returns
+    -------
+    str
+        The URL to the latest version of snaphu tarball if found, otherwise an empty string.
+
+    """
+    response = requests.get(SNAPHU_URL)
+    response.raise_for_status()  # Raise an HTTPError if the HTTP request returned an unsuccessful status code
+
+    # Parse the HTML content
+    soup = BeautifulSoup(response.text, "html.parser")
+    links = soup.find_all("a")
+
+    # Find all .tar.gz files
+    tarball_links = [link for link in links if link.get("href", "").endswith(".tar.gz")]
+
+    # Filter out the links that match the pattern "snaphu-vXXX.tar.gz"
+    snaphu_links = [
+        link for link in tarball_links if "snaphu-v" in link.get("href", "")
+    ]
+
+    if not snaphu_links:
+        return ""
```

Comment:
Let's raise an exception here instead.

```suggestion
        raise RuntimeError("failed to find any SNAPHU tarball links")
```